### PR TITLE
117 > Connect Test, Dashboard and Sign up 

### PR DESCRIPTION
**Pull request by thejoefriel**
#### Issues this PR relates to
#117

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

- Rebuild database
- Go through test
- Sign up 
- Check database
- Go back to your dashboard

I've also tried to cover a few edge cases:

If the user does the test but then leaves and comes back later
- Go through test
- don't log in 
- click around 
- go back to test landing - you should see now two options - one to return to results (should still show you then your results) and one to start quiz again (this will clear it from memory)

If user gets to dashboard and then wants to edit something in quiz
- As long as they don't go alway back to change the tasks (as this dictates the questions) then they can change answers

Also I've removed the links from sign up to log in and vice versa. The only reason you would sign up is to store your results (there's nothing else to have an account for) and this is then a lean way to ensure we're not having to validate against users who've already created an account then going on to do the test and also users who do the test again then try to log in with an existing account that's done the test etc. 

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/pages/Dashboard/Volunteer.js [hunk 1]
```diff
@@ -37,9 +50,37 @@ const VolunteerDashboard = () => {
...
+      const perc = Math.floor((hasNum / totalNum) * 100);
+
+      setSkills({
+        alreadyHasSkills: String(hasNum),
+        newlyCompleted: '0',
+        totalSkills: String(totalNum),
+      });
+      setScore({ assessmentScore: perc, improvementScore: 0 });
+    };
+    const assessment = getAssessmentFromStorage();
```
Comment (dupreesi) [inline client/src/pages/Dashboard/Volunteer.js:72]: can this be moved up (before the re-structure happens)?

```
 const {
        possibleSkills,
        skillsUserHas,
        skillsUserDoesntHave,
      } = assessment;     
```
 Or is this another assessment?

File: client/src/pages/Signup/Volunteer.js [hunk 2]
```diff
@@ -59,7 +63,42 @@ const VolunteerSignup = ({ user, onLogin, onLogout, onCreateAccount }) => {
...
+  const getAssessmentFromStorage = () => {
+    const assessment = JSON.parse(localStorage.getItem('assessment'));
+    if (assessment) {
+      return assessment;
+    }
+    return null;
+  };
+
   useEffect(() => {
+    const assessment = getAssessmentFromStorage();
```
Comment (dupreesi) [inline client/src/pages/Signup/Volunteer.js:75]: maybe you could create a helper function (e.g. manageUserAssessment) and use it in Dashboard and SignUp? Or are they doing different things? If so ignore me :)

**Review by dupreesi — CHANGES_REQUESTED** [review]
REALLY AWESOME STUFF! Amazing to see it all coming together and live :)

File: client/src/components/Menu/index.js [hunk 3]
```diff
@@ -3,14 +3,16 @@ import { useHistory } from 'react-router-dom';
 import { NAV_ROUTES } from '../../constants/nav-routes';
 import { useAuth } from '../../context/auth';
 import * as S from './style';
+// import { logout as logoutApi } from '../../api-calls/users';
```
[the remark this answers is not in the document]
Comment (thejoefriel) [reply client/src/components/Menu/index.js:6]: done

File: client/src/pages/Dashboard/Volunteer.js (hunk 1 above)
Comment (thejoefriel) [reply client/src/pages/Dashboard/Volunteer.js:72]: Cool moved it up a few lines 👍

File: client/src/pages/Signup/Volunteer.js (hunk 2 above)
Comment (thejoefriel) [reply client/src/pages/Signup/Volunteer.js:68]: done

**Review by dupreesi — APPROVED** [review]
